lewis_emulators/aldn1000/interfaces/stream_interface.py

```diff
@@ -0,0 +1,166 @@
+    def get_status(self, address):
+        return self.basic_get_response(address, self.device.status)
+
+    @if_connected
+    def set_pump_on(self, address):
+        self.device.address = address
+        self.device.pump = 'ON'
+
+    @if_connected
+    def set_pump_off(self, address):
+        self.device.address = address
+        self.device.pump = 'OFF'
+
+    @if_connected
+    def get_diameter(self, address):
+        return self.basic_get_response(address, self.format_data(self.device.diameter))
+
+    @if_connected
+    def set_diameter(self, address, diameter):
+        self.device.address = address
+        self.device.diameter = diameter
+        if diameter > 14.00:
+            self.device.volume_units = 'mL'
+        else:
+            self.device.volume_units = 'uL'
+        return self.basic_set_response(address)
+
+    @if_connected
+    def get_volume(self, address):
+        if self.device.volume_units == 'mL':
+            units = 'ML'
+        else:
+            units = 'UL'
+        return self.basic_get_response(address, self.format_data(self.device.volume), units)
+
+    @if_connected
+    def set_volume(self, address, volume):
+        self.device.address = address
+        self.device.volume = volume
+        return self.basic_set_response(address)
+
+    @if_connected
+    def get_volume_dispensed(self, address):
+        data = 'I{infused}W{withdrawn}{units}'.format(infused=self.format_data(self.device.volume_infused),
+                                                      withdrawn=self.format_data(self.device.volume_withdrawn),
+                                                      units=self.device.volume_units)
+        return self.basic_get_response(address, data)
+
+    @if_connected
+    def get_direction(self, address):
+        return self.basic_get_response(address, self.device.direction)
+
+    @if_connected
+    def set_direction(self, address, direction):
+        self.device.address = address
+        if direction == 'REV':  # Reverse
+            if self.device.direction == 'INF':  # Infuse
+                self.device.direction = 'WDR'  # Withdraw
+            else:
+                self.device.direction = 'INF'
+        else:
+            self.device.direction = direction
+        return self.basic_set_response(address)
+
+    @if_connected
+    def get_rate(self, address):
+        return self.basic_get_response(address, self.format_data(self.device.rate), self.device.units)
+
+    @if_connected
+    def set_rate(self, address, rate, units):
+        self.device.address = address
+        self.device.rate = rate
+        self.device.units = units
+        return self.basic_set_response(address)
+
+    @if_connected
+    def get_program_function(self, address):
+        return self.basic_get_response(address, self.device.program_function)
+
+    @if_connected
+    def clear_volume(self, address, volume_type):
+        self.device.address = address
+        if volume_type == 'INF':
```

Comment:
This logic should be in the device rather than the interface

Reply:
Moved into device.